app/models/plate/creator.rb

```diff
@@ -103,11 +117,19 @@ def create_plates(source_plate_barcodes, current_user, creator_parameters = nil)
           raise PlateCreationError, "Scanned plate #{scanned} has a purpose #{plate.purpose.name} not valid for creating [#{plate_purposes.map(&:name).join(',')}]"
         end
 
-        plates.concat(create_child_plates_from(plate, current_user, creator_parameters))
+        add_created_plates(plates, plate, create_child_plates_from(plate, current_user, creator_parameters))
```

Comment:
```suggestion
        plates.concat(add_created_plates(plate, create_child_plates_from(plate, current_user, creator_parameters)))
```

find it confusing passing the `plates` parameter to the `add_created_plates` method because it's only used in the plate scenario rather than the tub rack scenario. Can you do this instead, and make `add_created_plates` return `destinations`?

Reply:
ok